pkg/querier/frontend/frontend.go

```diff
@@ -99,6 +103,7 @@ func New(cfg Config, log log.Logger, limits *validation.Overrides) (*Frontend, e
 		log:    log,
 		queues: map[string]chan *request{},
 	}
+	f.cond = sync.NewCond(&f.mtx)
```

Comment:
This is just a move, used to be at bottom of the function.  Prefer it here, but can do in different PR if necessary.